site-builder/src/site/manager.rs

```diff
@@ -89,16 +89,20 @@ impl SiteManager {
                 true,
             ),
         };
+
+        // Handle existing resources separately to check their epochs and re-store if necessary
+        self.update_existing_resources(&existing_resources).await?;
```

Comment:
Important: this is probably not the right place to check if the existing resources need to be updated. Some of these resources may need to be removed anyways, so why reupload them? 

Suggestion: move this after computing the diff. in the diff, compute the set of "unchanged" resources, and only try to upload those. This way we minimize the necessary calls to walrus (and cost/latency of the operation)

Reply:
Ok, changed the approach as to the suggested one! With the only difference that I am implementing the "unchanged" resources computation in a different method than diff. Now, the resources from the unchanged set are now being pulled from the new resources set because I had trouble finding the full path in the existing resources set one. Is it better the calculation to be inside the diff method ? Additionaly, what is the difference between this approach and simply replacing everything from the beginning.